### PR TITLE
NO-JIRA: Ignore kubectl version mistmaches in Component Versions

### PR DESCRIPTION
**Pull request by ardaguclu**
**What this PR does / why we need it**:
This PR incorporates the changes in openshift/oc#1685 in Hypershift but with a slight 
differences. Since oc uses all the component versions, we are printing a warning to a user
about kubectl mismatches. On the other hand, Hypershift only deals with [version](https://github.com/openshift/hypershift/blob/b4209f3c421a319efa0a350737edc17318fd87ae/control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go#L2073) and
[kubernetes](https://github.com/openshift/hypershift/blob/e8a0c300a659d8d3cb0507b842771ee820a9554e/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L3954) fields in component versions and there is no benefit of a kubectl version mismatch warning. 
Besides, printing a warning would require parameter changes in functions to pass `ErrOut`.

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by ardaguclu** [comment]
/hold
until the actual PR in oc merges

**Comment by ardaguclu** [comment]
I don't think it is related;
/retest

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by ardaguclu** [comment]
@csrwng could you PTAL? (since you were the approver of the previous one https://github.com/openshift/hypershift/pull/3458). Thanks.

**Pull request merged**
